src/Runner.Listener/MessageListener.cs

```diff
@@ -319,7 +319,8 @@ private TaskAgentMessage DecryptMessage(TaskAgentMessage message)
                 var keyManager = HostContext.GetService<IRSAKeyManager>();
                 using (var rsa = keyManager.GetKey())
                 {
-                    return aes.CreateDecryptor(rsa.Decrypt(_session.EncryptionKey.Value, RSAEncryptionPadding.OaepSHA1), message.IV);
+                    var padding = _settings.RequireFipsCryptography ? RSAEncryptionPadding.OaepSHA256 : RSAEncryptionPadding.OaepSHA1;
```

Comment:
how does this work with Hosted Runner? i want to make sure we don't kill all hosted run when the runner rolls out.

Reply:
See my comment at the bottom of this PR. Its changed a bit since you last looked, but the gist is that we can safely upgrade forward here, as long as the server changes are deployed to support both styles with back compat, so I wont merge until Im sure server changes are rolled out everywhere, to prevent bad runner deploy